[2025-07-26T15:21:27.068+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-26T15:21:27.309+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T15:21:22.710871+00:00 [queued]>
[2025-07-26T15:21:27.321+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T15:21:22.710871+00:00 [queued]>
[2025-07-26T15:21:27.322+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-26T15:21:27.349+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_order_and_delivery_data> on 2025-07-26 15:21:22.710871+00:00
[2025-07-26T15:21:27.361+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=85) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T15:21:27.363+0000] {standard_task_runner.py:63} INFO - Started process 89 to run task
[2025-07-26T15:21:27.362+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sample_ETL', 'load_order_and_delivery_data', 'manual__2025-07-26T15:21:22.710871+00:00', '--job-id', '101', '--raw', '--subdir', 'DAGS_FOLDER/sample_ETL.py', '--cfg-path', '/tmp/tmp5t0erufa']
[2025-07-26T15:21:27.366+0000] {standard_task_runner.py:91} INFO - Job 101: Subtask load_order_and_delivery_data
[2025-07-26T15:21:27.442+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-26T15:21:27.563+0000] {task_command.py:426} INFO - Running <TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T15:21:22.710871+00:00 [running]> on host 88b2e53952eb
[2025-07-26T15:21:27.836+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sample_ETL' AIRFLOW_CTX_TASK_ID='load_order_and_delivery_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-26T15:21:22.710871+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-26T15:21:22.710871+00:00'
[2025-07-26T15:21:27.837+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-26T15:21:27.853+0000] {logging_mixin.py:188} INFO - ðŸš€ Starting ETL with JSON files
[2025-07-26T15:21:27.856+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:27.886+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T15:21:27.887+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:28.049+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:28.050+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:28.052+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:21:28.102+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'orders_topic'
[2025-07-26T15:21:28.103+0000] {logging_mixin.py:188} INFO - âœ… Orders producer completed
[2025-07-26T15:21:28.104+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:28.108+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T15:21:28.109+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:28.111+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('orders_topic',)
[2025-07-26T15:21:28.151+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:28.152+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:28.153+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:21:28.268+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_orders is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T15:21:28.271+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_orders
[2025-07-26T15:21:28.277+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T15:21:28.282+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_orders
[2025-07-26T15:21:28.285+0000] {base.py:419} INFO - Failed to join group etl_group_orders: NodeNotReadyError: coordinator-1
[2025-07-26T15:21:28.290+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:28.294+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:28.396+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T15:21:28.418+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-ad6ac3a7-d7eb-4ff9-8e22-6a93d79ab8d2 for group etl_group_orders; will retry join-group
[2025-07-26T15:21:28.420+0000] {base.py:419} INFO - Failed to join group etl_group_orders: [Error 79] MemberIdRequiredError
[2025-07-26T15:21:28.422+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T15:21:31.442+0000] {base.py:594} INFO - Successfully joined group etl_group_orders <Generation 9 (member_id: kafka-python-2.2.15-ad6ac3a7-d7eb-4ff9-8e22-6a93d79ab8d2, protocol: range)>
[2025-07-26T15:21:31.443+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T15:21:31.462+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('orders_topic', 0)]
[2025-07-26T15:21:31.464+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('orders_topic', 0),) for group etl_group_orders
[2025-07-26T15:21:43.619+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into orders
[2025-07-26T15:21:43.644+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T15:21:43.649+0000] {base.py:891} INFO - Leaving consumer group (etl_group_orders).
[2025-07-26T15:21:43.673+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_orders returned successfully
[2025-07-26T15:21:43.677+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:21:43.680+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T15:21:43.683+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:21:43.688+0000] {logging_mixin.py:188} INFO - âœ… Orders consumer completed
[2025-07-26T15:21:48.795+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'delivery_topic'
[2025-07-26T15:21:48.797+0000] {logging_mixin.py:188} INFO - âœ… Delivery producer completed
[2025-07-26T15:21:48.802+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:48.812+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T15:21:48.814+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:48.818+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('delivery_topic',)
[2025-07-26T15:21:48.907+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:48.909+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:48.910+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:21:49.013+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_delivery is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T15:21:49.014+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_delivery
[2025-07-26T15:21:49.015+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T15:21:49.017+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_delivery
[2025-07-26T15:21:49.017+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: NodeNotReadyError: coordinator-1
[2025-07-26T15:21:49.019+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T15:21:49.020+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T15:21:49.122+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T15:21:49.126+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-6c129011-6a2e-4617-82f4-47d291a05412 for group etl_group_delivery; will retry join-group
[2025-07-26T15:21:49.127+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: [Error 79] MemberIdRequiredError
[2025-07-26T15:21:49.128+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T15:21:52.134+0000] {base.py:594} INFO - Successfully joined group etl_group_delivery <Generation 9 (member_id: kafka-python-2.2.15-6c129011-6a2e-4617-82f4-47d291a05412, protocol: range)>
[2025-07-26T15:21:52.137+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T15:21:52.162+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('delivery_topic', 0)]
[2025-07-26T15:21:52.165+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('delivery_topic', 0),) for group etl_group_delivery
[2025-07-26T15:22:04.278+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into delivery
[2025-07-26T15:22:04.298+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T15:22:04.301+0000] {base.py:891} INFO - Leaving consumer group (etl_group_delivery).
[2025-07-26T15:22:04.363+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_delivery returned successfully
[2025-07-26T15:22:04.364+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:22:04.365+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T15:22:04.366+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T15:22:04.367+0000] {logging_mixin.py:188} INFO - âœ… Delivery consumer completed
[2025-07-26T15:22:04.368+0000] {logging_mixin.py:188} INFO - âœ… Task2 of ETL completed
[2025-07-26T15:22:04.369+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-26T15:22:04.370+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-26T15:22:04.419+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=sample_ETL, task_id=load_order_and_delivery_data, run_id=manual__2025-07-26T15:21:22.710871+00:00, execution_date=20250726T152122, start_date=20250726T152127, end_date=20250726T152204
[2025-07-26T15:22:04.472+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-26T15:22:04.541+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-26T15:22:04.554+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
